README.md

```diff
@@ -60,53 +60,60 @@ puts shipment
 
 ### Custom Connections
 
-Set `EasyPost.default_connection` to an object that responds to `call(method, path, api_key = nil, body = nil)`
+Pass in a lambda function as `custom_client_exec` when initializing a client that responds to `call(method, uri, headers, open_timeout, read_timeout, body = nil)` where:
+- `uri` is the fully-qualified URL of the EasyPost endpoint, including query parameters (`Uri` object)
+- `method` is the lowercase name of the HTTP method being used for the request (e.g. `:get`, `:post`, `:put`, `:delete`)
+- `headers` is a hash with all headers needed for the request pre-populated, including authorization (`Hash` object)
+- `open_timeout` is the number of seconds to wait for the connection to open (integer)
+- `read_timeout` is the number of seconds to wait for one block to be read (integer)
+- `body` is a string of the body data to be included in the request, or nil (e.g. GET or DELETE request) (string or `nil`)
+
+The lambda function should return an object with `code` and `body` attributes, where:
+- `code` is the HTTP response status code (integer) 
+- `body` is the response body (string)
 
 #### Faraday
 
 ```ruby
 require 'faraday'
-require 'faraday/response/logger'
-require 'logger'
-
-EasyPost.default_connection = lambda do |method, path, api_key = nil, body = nil|
-  Faraday
-    .new(url: EasyPost.api_base, headers: EasyPost.default_headers) { |builder|
-      builder.use Faraday::Response::Logger, Logger.new(STDOUT), {bodies: true, headers: true}
-      builder.adapter :net_http
-    }
-    .public_send(method, path) { |request|
-      request.headers['Authorization'] = EasyPost.authorization(api_key)
-      request.body = JSON.dump(EasyPost::InternalUtilities.objects_to_ids(body)) if body
-    }.yield_self { |response|
-      EasyPost.parse_response(
-        status: response.status,
-        body: response.body,
-        json: response.headers['Content-Type'].start_with?('application/json'),
-      )
-    }
-end
+
+custom_connection = lambda { |method, uri, headers, _open_timeout, _read_timeout, body = nil|
```

Comment:
praise: thanks for updating the readme, I will make the upgrade guide PR as ready for review